src/main.py

```diff
@@ -0,0 +1,79 @@
+import argparse
+
+import config
+import rds
+import tf
+import utils
+
+LOGGER = utils.setup_custom_logger("root")
+
+
+def main(command_line=None):
+    parser = argparse.ArgumentParser(description="Restore RDS snapshot")
+    parser.add_argument("-c", "--config", required=True)
+    parser.add_argument("-t", "--tfstate")
+    args = parser.parse_args(command_line)
+
+    data = config.load_config(args.config)
+    config.validate(data)
+
+    if data["Target"].get("AssumeRole") is not None:
+        LOGGER.info("Assuming target AWS role")
+        data["Target"]["AssumeRole"] = utils.assume_aws_role(
+            data["Target"]["AssumeRole"]
+        )
+    else:
+        data["Target"]["AssumeRole"] = None
```

Comment:
Couple of comments here:  
1. Generally the arguments we get from the outside - call to the program and parsed into data hash array should be read only.  We can create other variables to inject values into an pass to the code bellow.  So instead of setting `data["Target"]["AssumeRole"] = utils.assume_aws_role(data["Target"]["AssumeRole"])`  we could simply set `target_credentials = utils.assume_aws_role(data["Target"]["AssumeRole"])
2. Second point I would like to make is that the `utils.assume_aws_role` method is actually returning credentials.  Why not call it `get_aws_credentials`.  
3. Even more the method `get_aws_credentials` could then decide whether to create and return aws credentials if the role passed is not None, otherwise return None.  That way eliminating the whole if/else block above to simply:
`target_credentials = utils.get_aws_credentials(data["Target"]["AssumeRole"])`

To me it makes this code a lot simpler and readable when we want to maintain and reuse it.

Reply:
One more comment.  If you move the:
```python
    source = data["Source"]
    target = data["Target"]
```
above this code block then both this and the `data["Source"]["Share"]["AssumeRole"]` block bellow can be simply:
```python
    source = data["Source"]
    target = data["Target"]

    target_credentials = utils.get_aws_credentials(target["AssumeRole"])
    source_share_credentials = utils.get_aws_credentials(source["Share"])
```
Then anywhere you use `data["Target"]["AssumeRole"] ` you can replace it with `target_credentials` and same for  `data["Source"]["Share"]["AssumeRole"]` => ` source_share_credentials`